insights/specs/datasources/compliance/compliance_ds.py

```diff
@@ -0,0 +1,169 @@
+============================
+
+Only when option `--compliance` is specified, the insights collection will
+collect data for compliance.  When it is specified, only the Insights Spec
+:py:attr:`insights.specs.default.DefaultSpecs.compliance` will be
+collected. To collect other specs within the collection of compliance data,
+add them to the "persist" section in the "compliance_manifest"
+pre-defined in the :py:mod:`insights.specs.datasources.manifests`
+"""
+import logging
+import os
+import pkgutil
+import sys
+import yaml
+
+from tempfile import NamedTemporaryFile
+
+from insights.core.context import HostContext
+from insights.core.exceptions import SkipComponent
+from insights.core.plugins import datasource
+from insights.core.spec_factory import DatasourceProvider
+from insights.parsers.installed_rpms import InstalledRpms
+from insights.parsers.os_release import OsRelease
+from insights.parsers.redhat_release import RedhatRelease
+from insights.specs.datasources.compliance import ComplianceClient, SSG_PACKAGE, REQUIRED_PACKAGES
+
+# Since XPath expression is not supported by the ElementTree in Python 2.6,
+# import insights.contrib.ElementTree when running python is prior to 2.6 for compatibility.
+# Script insights.contrib.ElementTree is the same with xml.etree.ElementTree in Python 2.7.14
+# Otherwise, import defusedxml.ElementTree to avoid XML vulnerabilities,
+# if dependency not installed import xml.etree.ElementTree instead.
+if sys.version_info[0] == 2 and sys.version_info[1] <= 6:
+    import insights.contrib.ElementTree as ET
+else:
+    try:
+        import defusedxml.ElementTree as ET
+    except:
+        import xml.etree.ElementTree as ET
+
+
+logger = logging.getLogger(__name__)
+
+
+@datasource([OsRelease, RedhatRelease], HostContext)
+def os_version(broker):
+    os_release = None
+    if OsRelease in broker:
+        os_release = broker[OsRelease].get("VERSION_ID")
+    if not os_release and RedhatRelease in broker:
+        os_release = broker[RedhatRelease].version
+    if os_release:
+        return os_release.split('.')
+    raise SkipComponent('Cannot determine OS Version.')
+
+
+@datasource(InstalledRpms, HostContext)
+def package_check(broker):
+    rpms = broker[InstalledRpms]
+    missed = [rpm for rpm in REQUIRED_PACKAGES if rpm not in rpms]
+    if missed:
+        msg = 'Missing required packages for compliance scanning. Please ensure the following packages are installed: {0}\n'.format(', '.join(missed))
+        logger.error(msg)
+        raise SkipComponent(msg)
+
+    return rpms.newest(SSG_PACKAGE).version
+
+
+# timeout=0 disables the datasource timeout alarm
+# allowing compliance to run for as long as necessary
+@datasource(os_version, package_check, HostContext, timeout=0)
+def compliance(broker):
+    """
+    Custom datasource to collects content for Insights Compliance
+    """
+    try:
+        # Only run malware-detection if it was passed as an option to insights-client
```

Comment:
Guess mean to say "compliance" in this commented line?

Reply:
Fixed